src/shared/settingsConfiguration.ts

```diff
@@ -5,231 +5,590 @@
+
+            return false
+        }
+    }
+
+    /**
+     * Returns a "slice" of the settings configuration.
+     *
+     * The returned {@link Settings} interface operates against the provided section/scope.
+     */
+    public getSection(section: string, scope?: vscode.ConfigurationScope): Settings {
+        const parts = section.split('.')
+        const targetSection = parts.pop() ?? section
+        const parentSection = parts.join('.')
+
+        const resolveTarget = () => this.workspace.getConfiguration(section, scope)
+        const resolveParent = () => this.workspace.getConfiguration(parentSection, scope)
+
+        return {
+            get: (key: string, defaultValue?: unknown) => resolveTarget().get(key, defaultValue),
+            update: async (key: string, value: unknown) => {
+                // VS Code's settings API can read nested props but not write to them.
+                // We need to write to the parent if we cannot write to the child.
+                //
+                // TODO(sijaden): potentially provide valid keys upfront to avoid needing to
+                // handle this asymmetry with try/catch
+
+                try {
+                    return await resolveTarget().update(key, value, this.target)
+                } catch (error) {
+                    const parent = resolveParent()
+                    const val = parent.get(targetSection)
+
+                    if (typeof val === 'object') {
+                        return parent.update(targetSection, { ...val, [key]: value }, this.target)
+                    }
+
+                    throw error
+                }
+            },
+            reset: async () => {
+                const root = this.workspace.getConfiguration(undefined, scope)
+
+                return root.update(section, undefined, this.target)
+            },
         }
+    }
+
+    /**
+     * Creates a new {@link vscode.EventEmitter} that fires only for configuration changes affecting
+     * a particular section.
+     */
+    public createScopedEmitter(
+        section: string,
+        scope?: vscode.ConfigurationScope
+    ): [emitter: ScopedEmitter, listener: vscode.Disposable] {
+        const emitter: ScopedEmitter = new vscode.EventEmitter()
+        const toRelative = (sub: string) => (section ? [section, sub] : [sub]).join('.')
+
+        const listener = this.workspace.onDidChangeConfiguration(e => {
+            const affectsConfiguration = (section: string) => e.affectsConfiguration(toRelative(section), scope)
 
-        const val = settings.get<T>(settingKey)
-        return val ?? defaultValue
+            if (!section || e.affectsConfiguration(section, scope)) {
+                emitter.fire({ affectsConfiguration })
+            }
+        })
+
+        return [emitter, listener]
     }
 
     /**
-     * Gets the value of a VSCode setting.
+     * Defines a new set of configurables from the specified section.
```

Comment:
Let's avoid jargon if we can. I don't see much use for the current admixture of "settings" and "configuration" in the same module-family. It's just settings.

```suggestion
     * Defines a new set of settings from the specified section.
```